shotover-proxy/src/transforms/cassandra/sink_cluster/mod.rs

```diff
@@ -174,20 +173,47 @@ impl CassandraSinkCluster {
     }
 }
 
-fn create_query(messages: &Messages, query: &str, version: Version) -> Result<Message> {
+//Creating the AST is pretty expensive. Since we are defining the queries
+// ourselves and know they are not batch statements we can skip that check.
+// This function checks the messages list to ensure that the stream id used is not
+// already being used
+fn create_query_with_generated_stream_id(
+    messages: &Messages,
+    query: String,
+    version: Version,
+) -> Result<Message> {
     let stream_id = get_unused_stream_id(messages)?;
     Ok(Message::from_frame(Frame::Cassandra(CassandraFrame {
```

Comment:
Can we call create_query here instead of duplicating the logic?

Reply:
ahh, maybe we should just do something like `create_query(query, version, get_unused_stream_id(messages))` instead of having an entirely separate function for that.

Reply:
Good call.